sixtyfps_runtime/rendering_backends/gl/graphics_window.rs

```diff
@@ -534,26 +533,62 @@ impl PlatformWindow for GraphicsWindow {
         self.unmap_window();
     }
 
-    fn font_metrics(
+    fn text_size(
         &self,
-        item_graphics_cache_data: &corelib::item_rendering::CachedRenderingData,
+        item_graphics_cache: &corelib::item_rendering::CachedRenderingData,
         unresolved_font_request_getter: &dyn Fn() -> corelib::graphics::FontRequest,
-        reference_text: Pin<&Property<SharedString>>,
-    ) -> Box<dyn corelib::graphics::FontMetrics> {
+        text: &str,
+        max_width: Option<f32>,
+    ) -> Size {
         let font_request_fn = || {
             unresolved_font_request_getter().merge(&self.default_font_properties().as_ref().get())
         };
 
         let scale_factor =
             WindowProperties::FIELD_OFFSETS.scale_factor.apply_pin(self.properties.as_ref());
 
-        Box::new(super::fonts::FontMetrics::new(
+        crate::fonts::text_size(
             &self.graphics_cache,
-            item_graphics_cache_data,
+            item_graphics_cache,
             font_request_fn,
             scale_factor,
-            reference_text,
-        ))
+            text,
+            max_width,
+        )
+    }
+
+    fn text_input_byte_offset_for_position(
+        &self,
+        text_input: Pin<&sixtyfps_corelib::items::TextInput>,
+        pos: Point,
+    ) -> usize {
+        let scale_factor = self.scale_factor();
+        let cache_data = text_input
+            .cached_rendering_data
+            .get_or_update(&self.graphics_cache, || {
+                Some(crate::ItemGraphicsCacheEntry::Font(crate::fonts::FONT_CACHE.with(|cache| {
+                    cache.borrow_mut().font(
+                        text_input
+                            .unresolved_font_request()
+                            .merge(&self.default_font_properties.as_ref().get()),
+                        scale_factor,
+                        &text_input.text(),
```

Comment:
I guess this also needs a FIXME comment like in the other function?

Reply:
this adds a dependency.  so there is nothing to fix there.

Reply:
Ah yes, oops, I misread